`test_gc_of_remote_layers` is flaky

This appears to have destabilized a few days ago on the 24th of April:
![image](https://github.com/neondatabase/neon/assets/944640/ac8339ab-891b-48fe-8998-1bf476183629)

It's failing where the test expects _not_ to have seen the "Nothing to GC" message:

```
AssertionError: assert not ('2024-04-28T12:18:13.473590Z  INFO gc_loop{tenant_id=9766cca329bd50f96a6feb4351ec30eb shard_id=0000}:gc_timeline{timeline_id=234d138fdf10a77343e986d0eec4dee5 cutoff=0/0}: Nothing to GC: new_gc_cutoff_lsn 0/0, latest_gc_cutoff_lsn 0/149F060\n', LogCursor(_line_no=271))
 +  where ('2024-04-28T12:18:13.473590Z  INFO gc_loop{tenant_id=9766cca329bd50f96a6feb4351ec30eb shard_id=0000}:gc_timeline{timeline_id=234d138fdf10a77343e986d0eec4dee5 cutoff=0/0}: Nothing to GC: new_gc_cutoff_lsn 0/0, latest_gc_cutoff_lsn 0/149F060\n', LogCursor(_line_no=271)) = <bound method NeonPageserver.log_contains of <fixtures.neon_fixtures.NeonPageserver object at 0x7f9f3d3f1c40>>('Nothing to GC')
 +    where <bound method NeonPageserver.log_contains of <fixtures.neon_fixtures.NeonPageserver object at 0x7f9f3d3f1c40>> = <fixtures.neon_fixtures.NeonPageserver object at 0x7f9f3d3f1c40>.log_contains
 +      where <fixtures.neon_fixtures.NeonPageserver object at 0x7f9f3d3f1c40> = <fixtures.neon_fixtures.NeonEnv object at 0x7f9f3d3f1610>.pageserver
```

Potentially related change that tweaked pitr_interval behavior:
```
commit ee9ec26808d71e441b7d0c96bf9a046ced831f88
Author: John Spray <john@neon.tech>
Date:   Tue Apr 23 17:16:17 2024 +0100

    pageserver: change pitr_interval=0 behavior (#7423)
```

--- a/test_runner/regress/test_layer_eviction.py
+++ b/test_runner/regress/test_layer_eviction.py
@@ -159,6 +159,11 @@ def test_basic_eviction(
 def test_gc_of_remote_layers(neon_env_builder: NeonEnvBuilder):
     neon_env_builder.enable_pageserver_remote_storage(RemoteStorageKind.LOCAL_FS)
 
+    # Disable GC for all tenants, because we want to control when it runs.
+    # Without this, GC can sometimes run for initial tenant and fail the test.
+    # The test can fail because of unexpected "gc_loop{...}: Nothing to GC" log line.
+    neon_env_builder.pageserver_config_override = "tenant_config={gc_period='0s'}"
+
     env = neon_env_builder.init_start()
 
     tenant_config = {

--- a/test_runner/regress/test_layer_eviction.py
+++ b/test_runner/regress/test_layer_eviction.py
@@ -159,12 +159,9 @@ def test_basic_eviction(
 def test_gc_of_remote_layers(neon_env_builder: NeonEnvBuilder):
     neon_env_builder.enable_pageserver_remote_storage(RemoteStorageKind.LOCAL_FS)
 
-    # Disable GC for all tenants, because we want to control when it runs.
-    # Without this, GC can sometimes run for initial tenant and fail the test.
-    # The test can fail because of unexpected "gc_loop{...}: Nothing to GC" log line.
-    neon_env_builder.pageserver_config_override = "tenant_config={gc_period='0s'}"
-
-    env = neon_env_builder.init_start()
+    # don't create initial tenant, we'll create it manually with custom config
+    env = neon_env_builder.init_configs()
+    env.start()
 
     tenant_config = {
         "pitr_interval": "1s",  # set to non-zero, so GC actually does something

